drivers/gpio/gpio_hogs.c

```diff
@@ -90,16 +89,33 @@ static const struct gpio_hogs gpio_hogs[] = {
 	DT_FOREACH_STATUS_OKAY_NODE(GPIO_HOGS_COND_INIT)
 };
 
+static gpio_flags_t gpio_hogs_clear_flags_mask(void)
+{
+#ifdef CONFIG_GPIO_HOGS_MASK_OUTPUT
+	uint32_t reset_cause;
+
+	if ((hwinfo_get_reset_cause(&reset_cause) == 0) &&
+	    (reset_cause & CONFIG_GPIO_HOGS_RESET_CAUSE_MASK)) {
```

Comment:
Tying this to a reset cause mask feels a bit oddly specific. The idea of having the application code `gpio_hogs_init` may be good though. Maybe you could have an option for auto initializing the hog in Kconfig (similar to `CONFIG_USB_DEVICE_INITIALIZE_AT_BOOT`), and then an argument to `gpio_hogs_init` to skip level initialization? Then the application can initialize manually and use any logic for the argument.

Reply:
Good suggestion, let me try this out.